### PR TITLE
Added docker-push as dependency for create-tag

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: .github/workflows/compound-java.yml
Comment (OptimumCode) [reply .github/workflows/compound-java.yml:144]: I think we would need something like this:

`inputs.createTag && !cancelled() && (needs.docker-push.result == 'success' || needs.sonatype-push.result == 'success')`

The doc references are here:
https://docs.github.com/en/actions/learn-github-actions/expressions#cancelled
https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context

Comment (Nikita-Smirnov-Exactpro) [reply .github/workflows/compound-java.yml:144]: I would like to try my case. If it failure, we switch to your

Comment (OptimumCode) [reply .github/workflows/compound-java.yml:144]: Just want to mention that once you merge it to the main all other projects that use this workflow will be affected. I believe you could try with the dev branch you created.
But I will approve this PR anyway if you want to try after merging into the main branch

**Review by OptimumCode — APPROVED** [review]

**Comment by Nikita-Smirnov-Exactpro** [comment]
@OptimumCode you were right )
https://github.com/th2-net/.github/pull/38

**Pull request merged**
